### ISSUE TITLE
randperm default return type differs between python and jit

### ISSUE THREAD

**Issue by Bycob** [pytorch/pytorch#53015]
## 🐛 Bug

When no type is specified, `torch.randperm` returns a LongTensor. But after `torch.jit.script`ing the model, it returns a FloatTensor.
When we specify the type (e.g. torch.int64) it works fine.

## To Reproduce
```python
import torch

class MinimalExample(torch.nn.Module):
    def __init__(self):
        super(MinimalExample, self).__init__()

    def forward(self, x):
        perm = torch.randperm(x.numel())
        print(perm)
        return x[perm]

model = MinimalExample()
x = torch.rand(3)
# works fine
print(model(x))

jit_model = torch.jit.script(model)
# error!
print(jit_model(x))
```
Output:
```
tensor([2, 0, 1])
tensor([0.8209, 0.2563, 0.9951])
 2
 0
 1
[ CPUFloatType{3} ]
Traceback (most recent call last):
  File "torch_randperm_issue.py", line 19, in <module>
    print(jit_model(x))
  File "***/local/lib/python3.6/site-packages/torch/nn/modules/module.py", line 727, in _call_impl
    result = self.forward(*input, **kwargs)
RuntimeError: The following operation failed in the TorchScript interpreter.
Traceback of TorchScript (most recent call last):
  File "torch_randperm_issue.py", line 10, in forward
        perm = torch.randperm(x.numel())
        print(perm)
        return x[perm]
               ~~~~~~~ <--- HERE
RuntimeError: tensors used as indices must be long, byte or bool tensors
```

## Expected behavior

Tensor `perm` should be of type `CPULongType` instead of `CPUFloatType` in the second case.

## Environment

```
PyTorch version: 1.7.1
Is debug build: False
CUDA used to build PyTorch: 10.2
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.5 LTS (x86_64)
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: 6.0.0-1ubuntu2 (tags/RELEASE_600/final)
CMake version: version 3.18.4

Python version: 3.6 (64-bit runtime)
Is CUDA available: True
CUDA runtime version: Could not collect
GPU models and configuration: 
GPU 0: TITAN X (Pascal)
GPU 1: GeForce GTX 1080 Ti
GPU 2: GeForce GTX 1080 Ti

Nvidia driver version: 455.23.05
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.7.6.5
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.0.5
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.0.5
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.0.5
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.0.5
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.0.5
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.0.5
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.0.5
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.19.5
[pip3] torch==1.7.1
[pip3] torchfile==0.1.0
[pip3] torchvision==0.8.2
[pip3] torchviz==0.0.1
```

cc @gmagogsfm

**Comment by gmagogsfm** [comment]
In eager mode, `torch.randperm` returns a tensor of dtype `torch.int64`, but in TS, it returns float tensor.

**Comment by gmagogsfm** [comment]
This has been fixed by #57105 closing
